etna/whatson/models.py

```diff
@@ -308,9 +317,18 @@ def get_context(self, request, *args, **kwargs):
         filter_form = EventFilterForm(request.GET)
 
         if filter_form.is_valid():
-            self.events = self.filter_form_data(filter_form.cleaned_data)
-
+            filtered_events = self.filter_form_data(filter_form.cleaned_data)
+            if self.featured_event:
+                context["events"] = filtered_events.exclude(id=self.featured_event_id)
+            else:
+                context["events"] = filtered_events
+        else:
+            context["events"] = self.events
         context["filter_form"] = filter_form
+        if self.featured_event:
+            context["show_fetured_event"] = self.should_show_featured_event(
+                filtered_events, self.featured_event
```

Comment:
Issue: in the event that `filter_form.is_valid()` evaluates to `False`, `filtered_events` will not have been declared, so this will cause an exception.

Reply:
Is it best to just declare `filtered_events` to equal `self.events` in that scenario? I'm not sure if the form can be invalid as it is a simple get.